### DISCUSSION TITLE
[SOLVED] Allow the Definition of Additional Events for the models

### DISCUSSION THREAD

**Discussion by ssmusoke** [filamentphp/filament#532] [Ideas]
If I use filament to edit a model that may be from another package, then the events from that package are lost

I woud like to be able to define additional events apart from the model on the different crud actions 

Any advice on how to achieve that too in the current usage would be appreciated

**Comment by ryangjchandler** [comment]
@ssmusoke I'm not sure what you're trying to achieve, have you got an example?

**Reply by ssmusoke** [reply]
I am editing a setting from https://github.com/spatie/laravel-settings via filament and have a listener for the SavedSettings which is not triggered when I edit the value

**Reply by ssmusoke** [reply]
@ryangjchandler any advice on how to emit such events?

**Reply by zhanang19** [reply]
You can use hooks

Ref: https://filamentadmin.com/docs/resources/#forms

**Reply by ssmusoke** [reply]
@zhanang19 Thanks this is what I am looking for.
